src/Components/ImportModal/ImportModal.tsx

```diff
@@ -12,36 +19,188 @@ type Props = {
 export const ImportModal: React.FC<Props> = ({
   setShow,
   removeFile,
-  addFile,
+  addFile
 }) => {
-  const createMediaElement = (files: File[]) => {
+  const [onDragState, setOnDragState] = useState<boolean>(false);
+  const [onDropState, setOnDropState] = useState<boolean>(false);
+  const [mediaReady, setMediaReady] = useState<number>(0);
+  const [medias, setMedias] = useState<Media[]>([]);
+  const removeMedia = (index: number): void => {
+    const newMedias = [...medias];
+    removeFile(index);
+    if (index > -1) {
+      newMedias.splice(index, 1);
+    }
+    setMedias(newMedias);
```

Comment:
With the current set up, you're now maintaining two lists of essentially the same thing: `files` (in App.tsx) and `medias` here.
That means that, as you've done here, to add or remove a file, you have to alter both arrays, rather than maintaining one source of truth.

Can you pass in the `files` list to this component, and then call `removeFiles` rather than `removeMedia`. Your component should then respond to `files` changes, and generate the required media elements from there.

The `files` currently aren't being used anywhere in their raw state: only `Media`. Will they be used in the future? What is the purpose of each array? How can we simplify so there's just one set of data?

Let's loop back on this - I think there's a better solution, something along the lines of `videoPresenter`